templates/auth.conf.erb

```diff
@@ -84,11 +84,13 @@
 path /puppet/v3/environments
 method find
 allow *
+<% if @puppetversion.to_f < 5.0 -%>
```

Comment:
That's the legacy API. On newer puppetservers (also 2.x), the proxy should use `/environment_classes` (just below in the file you pasted), but maybe @witlessbird can chime in here...